### PR TITLE
Fixes and refactoring for TODOList module

### PR DESCRIPTION
**Pull request by HeinzUBC**
Fixed invalid comment syntax sideBarView class in App.css.

Refactored code for TODOList components and redux toolkit files so that code is more concise.

Added addCategory reducer action to todoListSlice.js so that user-made categories will appear in category filter tool of TODOList module.

### CODE REVIEW

File: src/components/TODOList/AddNewTODO.js [hunk 1]
```diff
@@ -1,127 +1,130 @@
-import React, { useState } from 'react';
-import {useDispatch, useSelector} from "react-redux";
-import DatePicker from "react-datepicker";
-import {Button, Card, Dialog, Input, Textarea} from "@material-tailwind/react";
-import {addTODO} from "../../store/TODOList_Slice";
+import React, {useState} from 'react';
```
Comment (awctw) [inline src/components/TODOList/AddNewTODO.js:1]: Is your linter not turned on?

Comment (HeinzUBC) [reply src/components/TODOList/AddNewTODO.js:1]: I turned it on I think

Comment (awctw) [reply src/components/TODOList/AddNewTODO.js:1]: Ok I will just add it in the repo in the next merge so its always on.

**Review by awctw — APPROVED** [review]

File: src/components/TODOList/AddNewTODO.js (hunk 1 above)
Comment (HeinzUBC) [reply src/components/TODOList/AddNewTODO.js:1]: Thank you very much. It seems that prettier does not work automatically in webstorm. I think I have a click "reformat code" explicitly to trigger it? I guess in something like Vscode it runs instantly?

Comment (awctw) [reply src/components/TODOList/AddNewTODO.js:1]: You have to change your settings for it to be automatic everything the code is saved

**Pull request merged**